core/src/main/java/org/ehcache/spi/cache/tiering/CachingTier.java

```diff
@@ -47,6 +48,14 @@
   void invalidate(K key) throws CacheAccessException;
 
   /**
+   * Remove a mapping then call a function.
+   * @param key the key.
+   * @param invalidator the function to call.
+   * @throws CacheAccessException
+   */
+  void invalidate(K key, NullaryFunction<K> invalidator) throws CacheAccessException;
+
+  /**
    * Empty out the caching store.
    * @throws CacheAccessException
    */
```

Comment:
I thought this was being handled in a different issue but apparently I was wrong. I'll rename it.